### ISSUE TITLE
Java RESTException enhancements

### ISSUE THREAD

**Issue by johnmac-ibm** [IBMStreams/streamsx.topology#1068]
Couple of enhancements requested for the RESTException in the Java API in v1_7_experimental branch:

1. `getStatusCode()` appears to only be set for 404 errors, it would be nice if it was set for other HTTP errors as well (even if those don't have Streams-specific message)
2. When testing with invalid service URLs it sometimes threw an `IllegalStateException` from the Gson code. It appears some URLs will return a 404 but not with a JSON object in the response:

```
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:187)
	at com.google.gson.Gson.fromJson(Gson.java:805)
	at com.google.gson.Gson.fromJson(Gson.java:770)
	at com.google.gson.Gson.fromJson(Gson.java:719)
	at com.google.gson.Gson.fromJson(Gson.java:691)
	at com.ibm.streamsx.rest.RESTException.create(RESTException.java:31)
	at com.ibm.streamsx.rest.StreamsConnection.getResponseString(StreamsConnection.java:115)
	at com.ibm.streamsx.rest.StreamsConnection.getInstance(StreamsConnection.java:158)
```

It would be nice if instead of that it still threw a `RESTException` with the code, even if it didn't include the enhanced JSON message.

**Comment by yvchan** [comment]
i'm working on this.

**Comment by yvchan** [comment]
this has been fixed in pr #1070
